segments/stack_project.p9k

```diff
@@ -0,0 +1,34 @@
+# vim:ft=zsh ts=2 sw=2 sts=2 et fenc=utf-8
+################################################################
+# @title powerlevel9k Segment - Haskell Stack Project
+# @source [powerlevel9k](https://github.com/bhilburn/powerlevel9k)
+##
+
+################################################################
+# Register segment
+# Parameters:
+#   segment_name  context  foreground  background  Generic  Flat/Awesome-Patched  Awesome-FontConfig  Awesome-Mapped-FontConfig  NerdFont
+#                                                                                    
+p9k::register_segment "STACK_PROJECT" "" "purple3" "white" ''  $'\u03BB='  $'\u03BB='  $'\u03BB='  $'\uE777'
+
+################################################################
+# @description
+#   This segment displays if the current directory belongs to a Haskell Stack project.
+##
+# @args
+#   $1 string Alignment - left | right
+#   $2 integer Segment index
+#   $3 boolean Whether the segment should be joined
+##
+prompt_stack_project() {
+
+    local haskellstack_version=$(stack --version 2>/dev/null | grep -oE '[0-9.]+' | head -n1)
+    local stackyaml_file="stack.yaml"
```

Comment:
Is there a reason, why that is a separate variable? It is used only once in the code..

Reply:
No reason whatsoever, I have changed it so `__p9k_upsearch` uses the string directly without storing it in any variable. Thanks!